### PR TITLE
Lj eps prefactor

### PR DESCRIPTION
**Pull request by maxentile**
restores lj eps special case from https://github.com/proteneer/timemachine/pull/685 partially overwritten in https://github.com/proteneer/timemachine/pull/931 (20x storage difference when it is not a requirement to be able to vary ligand LJ sig)

(refactoring was attempted for a couple hours in https://github.com/proteneer/timemachine/pull/1210 , might be something to revisit in the future)

### CODE REVIEW

File: timemachine/potentials/nonbonded.py [hunk 1]
```diff
@@ -562,6 +562,93 @@ def coulomb_interaction_group_energy(q_ligand: Array, q_prefactors: Array) -> fl
...
+        LJ_sig of all other atoms (in environment)
+    eps_others: [N_env] array
+        LJ_eps of all other atoms (in environment)
+    box: optional diagonal [D, D] array
+    cutoff: float
+
+    Returns
+    -------
+    prefactor_i : float
+        sum_j 4 * sqrt(eps_j) * ((sig_ij/r_ij)**12 - (sig_ij/r_ij)**6)
```
Comment (jkausrelay) [inline timemachine/potentials/nonbonded.py:593]: sqrt(eps_j) or just eps_j?

Comment (maxentile) [reply timemachine/potentials/nonbonded.py]: adjusted in https://github.com/proteneer/timemachine/pull/1212/commits/40ee31c9a647c1a1b4cc687dcf6cd2a5024f0be9

File: timemachine/potentials/nonbonded.py [hunk 2]
```diff
@@ -562,6 +562,93 @@ def coulomb_interaction_group_energy(q_ligand: Array, q_prefactors: Array) -> fl
...
 
 # utilities for efficiently recomputing energy as a function of ligand LJ parameters
+
+# note: two cases:
+# (1) only varying LJ eps parameters (can summarize using 1 float per ligand atom)
+# (2) simultaneously varying LJ eps and sig parameters (summarized using 20 floats per ligand atom)
+
+
+# case (1) : only varying ligand LJ eps parameters
+def lj_eps_prefactor_on_atom(x_i, x_others, sig_i, sig_others, eps_others, box=None, cutoff=1.2) -> float:
```
Comment (jkausrelay) [inline timemachine/potentials/nonbonded.py:572]: In some of the prefactor code this defaults to cutoff=np.inf vs cutoff=1.2 . Is there a reason for this?

Comment (maxentile) [reply timemachine/potentials/nonbonded.py]: oops, mistakenly used 1.2 in this PR, reverted to np.inf in https://github.com/proteneer/timemachine/pull/1212/commits/d5a57e56f89bfa501dc826ea37aebeee25a8e5e1

**Review by jkausrelay — APPROVED** [review]

**Pull request merged**